plugin/storage/cassandra/spanstore/dbmodel/converter.go

```diff
@@ -105,20 +118,38 @@ func (c converter) toDomain(dbSpan *Span) (*model.Span, error) {
 		StartTime:     model.EpochMicrosecondsAsTime(uint64(dbSpan.StartTime)),
 		Duration:      model.MicrosecondsAsDuration(uint64(dbSpan.Duration)),
 		Tags:          tags,
+		Warnings:      warnings,
 		Logs:          logs,
 		Process:       process,
 	}
 	return span, nil
 }
 
 func (c converter) fromDBTags(tags []KeyValue) ([]model.KeyValue, error) {
-	retMe := make([]model.KeyValue, len(tags))
-	for i := range tags {
-		kv, err := c.fromDBTag(&tags[i])
+	var retMe []model.KeyValue
+	for _, tag := range tags {
+		kv, err := c.fromDBTag(&tag)
```

Comment:
please change it back. Original version has no copying and no memory allocations, yours has both. And pre-allocating array of known size is usually more efficient because it avoids re-allocation as the vector grows.

Reply:
because your input and output lengths may not batch, you have to allocate with `make(type, 0, len)` and use `append`, not indexed assignment.

Reply:
done